### PR TITLE
feat: add endpoints, middleware, Prisma & MongoDB to users service

### PR DESCRIPTION
**Pull request by sheimoria**
# Changes
- [x] Add endpoints (`GET`, `POST`, `PUT`, `DELETE` users)
- [x] Add middleware (`cors`, `helmet`, `logger`, `swagger`)
- [x] Add Prisma ORM with `User` and `QuestionAttempt` schema
- [x] Connect to MongoDB Atlas with `DATABASE_URL` env var
- [x] Add `DATABASE_URL` env var to Pulumi config
- [x] Add Prettier and ESLint config

### CODE REVIEW

File: backend/users-service/app/src/middlewares/users.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { logger } from "@bogeychan/elysia-logger";
+import cors from "@elysiajs/cors";
+import jwt from "@elysiajs/jwt";
+import swagger from "@elysiajs/swagger";
+import Elysia from "elysia";
+import { helmet } from "elysia-helmet";
+
+export const middleware = new Elysia()
+    .use(cors())
```
Comment (raynerljm) [inline backend/users-service/app/src/middlewares/users.middleware.ts]: can specify the FE origin

File: backend/users-service/app/src/middlewares/users.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    .use(helmet())
+    .use(
+        jwt({
```
Comment (raynerljm) [inline backend/users-service/app/src/middlewares/users.middleware.ts]: as discussed can remove

File: backend/users-service/app/src/controllers/users.controller.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { CreateUserBody, UpdateUserBody } from "../models/user.model";
+import { create, read, remove, update } from "../services/users.service";
+
+export const del = ({ params: { id } }: { params: { id: string } }) => remove(id)
+
+export const get = ({ params: { id } }: { params: { id: string } }) => read(id)
+
+export const post = ({ body }: { body: CreateUserBody }) => create(body)
```
Comment (raynerljm) [inline backend/users-service/app/src/controllers/users.controller.ts]: can add a `TODO` to do role validation (need to discuss the implementation)

but idea is that a non-admin user should not be able to make themselves or others admin

File: backend/users-service/infra/Pulumi.prod.yaml [hunk 4]
```diff
@@ -2,13 +2,17 @@ config:
...
   users-service-infra:apiGatewayAuthSecret:
     secure: AAABALuikXMmWcgtFjbcEK7Mn+sh0PaKT2ko9ofpMSb8MR/TLIfKENuRuHK1B97Ypzq1dPjQARsJfwrKepRJQnOW32+4046TpXh7KHrZl6E8Eo1aJQaxPbMKcuJ4877GDDxJEHDLj3q+bShqczzYWNzGIrsQEwG9jWt9LwN8UrpnM8YEwAlpxPkI3APN63YcH5bvdi9/f154aF2Sxkb2Qjl9lNhuq6coOwXNzk3/18W9fJVD6JkE2ngZTNIcQFy/
-  users-service-infra:containerPort: "80"
-  users-service-infra:cpu: "256"
+  users-service-infra:containerPort: '80'
+  users-service-infra:cpu: '256'
   users-service-infra:hostedZoneId:
     secure: AAABAJI7xUrzlmZP58/kzcoW/P52p4r3p6/UZSSr3DhDIms5zaLioKYT0uBjRr+JxyxYHg==
-  users-service-infra:memory: "128"
+  users-service-infra:jwtSecret:
```
Comment (raynerljm) [inline backend/users-service/infra/Pulumi.prod.yaml]: as discussed can remove this

File: backend/users-service/infra/Pulumi.prod.yaml [hunk 5]
```diff
@@ -2,13 +2,17 @@ config:
   aws:region: ap-southeast-1
   users-service-infra:acmEcsCertificateArn:
     secure: AAABAMz6p82vt5eZ3hLTjpX3tAib3lPPeEBW14jvvFpgNnUwdR6ah+fN34lK+4uUDwOfPYZ4peYOR2shE5pPAGtYmIvOnyVPMWbsOiC1C+9kTLLXVqpvVYVE0hDwohsvhCRZ0iGK5YrGVtIfQhq2stzuyExh1GMC
+  users-service-infra:databaseUrl:
```
Comment (raynerljm) [inline backend/users-service/infra/Pulumi.prod.yaml:5]: as discussed - up to u whether u want to pass connection string or init atlas MongoDB within pulumi

File: backend/users-service/infra/Pulumi.staging.yaml [hunk 6]
```diff
@@ -4,11 +4,15 @@ config:
...
   users-service-infra:apiGatewayAuthSecret:
     secure: AAABAN1ckGUj2vV6q+gfqx3K21py4BqcpWiRyOlYQSdCoKNUkl8/UIESJs9gx5wBeZqQO6m3ToCBYNq5ec+mJWRFN+d6zsWrqoScre6MUbTORFSVr63O+4yScQMSofwHAr1D5LslAMmBd2Ycp8K226kwsBNt5We2PQkv9fTZrzBlKTbffdcJI0GDWCUrGqcIoCEy34b9h+uq+r8JPvx6xaa2xD0TCPtYJNw1/SBYa9NSW6Dr/7f8CvP8KzTZi7TW
-  users-service-infra:containerPort: "80"
-  users-service-infra:cpu: "256"
+  users-service-infra:containerPort: '80'
+  users-service-infra:cpu: '256'
   users-service-infra:hostedZoneId:
     secure: AAABAAqNubrYHxuf6vI2fqAeJOPUDnSjV7a3G4rYgDeFRgu3rnS2ruA4YZK0rIyYD4dueQ==
-  users-service-infra:memory: "128"
+  users-service-infra:jwtSecret:
```
Comment (raynerljm) [inline backend/users-service/infra/Pulumi.staging.yaml]: as discussed can remove this

File: backend/users-service/.prettierrc.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (raynerljm) [inline backend/users-service/.prettierrc.json:1]: as discussed can standardize the prettier w the one in the fe

File: backend/users-service/infra/index.ts [hunk 8]
```diff
@@ -1,107 +1,110 @@
...
-const acmCertificateArn = config.requireSecret("acmEcsCertificateArn");
+const acmCertificateArn = config.requireSecret('acmEcsCertificateArn')
 
 // For Auth module
 const apiGatewayAuthSecret = config.requireSecret("apiGatewayAuthSecret");
 
 const currentEnv = pulumi.getStack(); // 'staging' or 'prod'
 const isProd = currentEnv === "prod";
 
+const jwtSecret = config.requireSecret('jwtSecret')
```
Comment (raynerljm) [inline backend/users-service/infra/index.ts]: as discussed can remove

**Review by raynerljm — CHANGES_REQUESTED** [review]
few changes requested as discussed

File: backend/users-service/app/src/middlewares/users.middleware.ts (hunk 2 above)
Comment (sheimoria) [reply backend/users-service/app/src/middlewares/users.middleware.ts]: Resolved in 5139e3bb0fe3d0d3edf4e402504878d4b588f9f6

File: backend/users-service/infra/Pulumi.prod.yaml (hunk 4 above)
Comment (sheimoria) [reply backend/users-service/infra/Pulumi.prod.yaml]: Resolved in 5139e3bb0fe3d0d3edf4e402504878d4b588f9f6

File: backend/users-service/infra/Pulumi.staging.yaml (hunk 6 above)
Comment (sheimoria) [reply backend/users-service/infra/Pulumi.staging.yaml]: Resolved in 5139e3bb0fe3d0d3edf4e402504878d4b588f9f6

File: backend/users-service/infra/index.ts (hunk 8 above)
Comment (sheimoria) [reply backend/users-service/infra/index.ts]: Resolved in 5139e3bb0fe3d0d3edf4e402504878d4b588f9f6

File: backend/users-service/app/src/middlewares/users.middleware.ts (hunk 1 above)
Comment (sheimoria) [reply backend/users-service/app/src/middlewares/users.middleware.ts]: Resolved in 5139e3bb0fe3d0d3edf4e402504878d4b588f9f6

File: backend/users-service/app/src/controllers/users.controller.ts (hunk 3 above)
Comment (sheimoria) [reply backend/users-service/app/src/controllers/users.controller.ts]: Resolved in 85385a3e713193a3f528369b85496ea88edc2ff9

File: backend/users-service/.prettierrc.json (hunk 7 above)
Comment (sheimoria) [reply backend/users-service/.prettierrc.json:1]: Resolved in d8c4ab7ca0323cf0dc91eba8a543f8af56febf72

**Review by raynerljm — APPROVED** [review]
lgtm! 🎉

**Pull request merged**